src/lib/operators/import.cpp

```diff
@@ -0,0 +1,82 @@
+#include "import.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+#include "hyrise.hpp"
+#include "import_export/binary/binary_parser.hpp"
+#include "import_export/csv/csv_parser.hpp"
+#include "utils/assert.hpp"
+#include "utils/load_table.hpp"
+
+namespace opossum {
+
+Import::Import(const std::string& filename, const std::optional<std::string>& tablename, const ChunkOffset chunk_size,
+               const FileType type, const std::optional<CsvMeta>& csv_meta)
+    : AbstractReadOnlyOperator(OperatorType::Import),
+      _filename(filename),
+      _tablename(tablename),
+      _chunk_size(chunk_size),
+      _type(type),
+      _csv_meta(csv_meta) {}
+
+const std::string& Import::name() const {
+  static const auto name = std::string{"Import"};
+  return name;
+}
+
+std::shared_ptr<const Table> Import::_on_execute() {
+  if (_tablename && Hyrise::get().storage_manager.has_table(*_tablename)) {
+    return Hyrise::get().storage_manager.get_table(*_tablename);
+  }
+
+  // Check if file exists before giving it to the parser
+  std::ifstream file(_filename);
+  Assert(file.is_open(), "Import: Could not find file " + _filename);
+  file.close();
+
+  const auto table = _import();
+
+  if (_tablename) {
+    Hyrise::get().storage_manager.add_table(*_tablename, table);
+  }
+
+  return table;

```

Comment:
I guess this is kind of related to the adding to storage manager thing. By letting the operator have an output, one could just read a table without any side effects in the past. Since we now have parsers we can call instead, there is no real reason for returning the table anymore.